eloquentjs/14_1_build_a_table/index.js

```diff
@@ -0,0 +1,41 @@
+const MOUNTAINS = [
+    {name: "Kilimanjaro", height: 5895, place: "Tanzania"},
+    {name: "Everest", height: 8848, place: "Nepal"},
+    {name: "Mount Fuji", height: 3776, place: "Japan"},
+    {name: "Vaalserberg", height: 323, place: "Netherlands"},
+    {name: "Denali", height: 6168, place: "United States"},
+    {name: "Popocatepetl", height: 5465, place: "Mexico"},
+    {name: "Mont Blanc", height: 4808, place: "Italy/France"}
+];
+
+const mountainsTableElement = generateObjectTable(MOUNTAINS);
+document.getElementById("mountains").appendChild(mountainsTableElement);
+
+function generateObjectTable(inputArray) {
+    const table = document.createElement('table');
+
+    const headerRow = document.createElement('tr');
+    Object.getOwnPropertyNames(inputArray[0]).forEach(propertyName => {
+        const cell = document.createElement('th')
+        cell.innerHTML = propertyName;
+        headerRow.appendChild(cell);
+    });
+    table.appendChild(headerRow);
+
+    inputArray.forEach(entry => {
```

Comment:
Consider replacing this with a for of. It's generally easier to visually parse, performs better, and is more idiomatic js. map and stuff are still great, it's specifically forEach that brings very little to the table.